woocommerce/woocommerce
# Fix Payments tab tracking

### All Submissions:






### Changes proposed in this Pull Request:



Closes #28603.

This PR adds the code necessary to track the Payments tab.
To save the Payments settings we [do the action `woocommerce_update_options_payment_gateways_[gateway_id]`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways.php#L230) that triggers [the method `process_admin_options`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/abstracts/abstract-wc-settings-api.php#L204-L220). That method is being hooked in every payment gateway (e.g. [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/gateways/cheque/class-wc-gateway-cheque.php#L44)).
Because we are saving the Payments settings in this way is why the [Tracks event `wcadmin_settings_change`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php#L102) isn't being triggered.
The changes in this PR allow us to trigger the mentioned Tracks event after the Payments settings are updated.

### How to test the changes in this Pull Request:

0. Check out this branch.
1. The easiest way to test this PR is using the [WooCommerce Admin Test Helper](https://github.com/woocommerce/woocommerce-admin-test-helper). Install and activate it. Be sure that [this PR](https://github.com/woocommerce/woocommerce-admin-test-helper/pull/25) has been merged, otherwise, you should set up the project and check out the branch `fix/8199`. 
2. Go to `WooCommerce` -> `Status` -> `Logs` (URL `wp-admin/admin.php?page=wc-status&tab=logs`).
3. Select a `tracks-*` log from the dropdown.
4. Go to `WooCommerce` -> `Settings` -> `Payments` (URL `/wp-admin/admin.php?page=wc-settings&tab=checkout`) and toggle the different Payment options.

![screenshot-woocommerce local-2022 02 04-15_13_34](https://user-images.githubusercontent.com/1314156/152581603-769f6b9b-c12f-407f-876d-e1e7c2a3d399.png)

Under the Log tab (mentioned on step 2), you should see:
```
DEBUG wcadmin_settings_change
DEBUG   - settings: woocommerce_[option]_settings
DEBUG   - tab: checkout
```

5. Verify that the other settings options (not related to the Payments tab) are working correctly.

Please notice that the `Save changes` button is not being tracked, since the payment methods toggle is saving and tracking every event individually.

![screenshot-woocommerce local-2022 02 04-15_13_34-2](https://user-images.githubusercontent.com/1314156/152588488-f7a5f897-6631-451e-837c-463ed42e9b7a.png)

Actually, I wonder if it might be worth removing completely in a follow-up PR.

### Other information:




### Changelog entry

Fix Payments tab tracking.

### FOR PR REVIEWER ONLY:
